apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/next/codegen/Common.kt

```diff
@@ -0,0 +1,409 @@
+package com.apollographql.apollo.compiler.next.codegen
+
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.api.internal.ResponseFieldMapper
+import com.apollographql.apollo.api.internal.ResponseFieldMarshaller
+import com.apollographql.apollo.api.internal.ResponseReader
+import com.apollographql.apollo.compiler.applyIf
+import com.apollographql.apollo.compiler.codegen.kotlin.KotlinCodeGen
+import com.apollographql.apollo.compiler.next.ast.CodeGenerationAst
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.BOOLEAN
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.DOUBLE
+import com.squareup.kotlinpoet.FLOAT
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.INT
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.LONG
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.asClassName
+import com.squareup.kotlinpoet.asTypeName
+import com.squareup.kotlinpoet.joinToCode
+
+internal fun CodeGenerationAst.FieldType.asTypeName(nullable: Boolean = false): TypeName {
+  return when (this) {
+    is CodeGenerationAst.FieldType.Scalar -> when (this) {
+      is CodeGenerationAst.FieldType.Scalar.ID -> ClassName.bestGuess(type)
```

Comment:
Importing these classes would make the code easier to read

Reply:
If we do that, I wouldn't be against some kind of prefixing like `ASTFieldType`, `IRFieldType`, etc... Maybe `FieldType` isn't too much used in other places but we have 7 different `Field` classes and it's sometimes hard to get which layer is used when opening a new file.

Reply:
I now realized that this is in Common.kt. So my suggestion is not a good idea.